File: ios/Kurir/modules/tasks/impl/TasksModule.swift
//
//  FindModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class TasksModule: Module {
    override func inject() {
        // ViewModel
        factory(TasksViewModel.self) { r in
            TasksViewModel(userData: r.resolve(UserData.self),
                           usersRepository: r.resolve(UsersRepository.self)!,
                           tasksRepository: r.resolve(TasksRepository.self)!)
        }
        
        // Delegate
        factory(TasksDelegate.self) { (r, view: TasksView) -> TasksDelegate in
            TasksDelegate(view: view, viewModel: r.resolve(TasksViewModel.self)!)
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/tasks/impl/TasksModule.kt
package au.com.crazybean.mobilex.kurir.modules.tasks.impl

import au.com.crazybean.mobilex.kurir.modules.tasks.TasksDelegate
import au.com.crazybean.mobilex.kurir.modules.tasks.TasksView
import au.com.crazybean.mobilex.kurir.modules.tasks.TasksViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val tasksModule = module {
    // ViewModel
    viewModel {
        TasksViewModel(get(), get(), get())
    }

    // Delegate
    factory { (view: TasksView?) ->
        TasksDelegate(view, get())
    }
}